backend/dbmigrations_handler.py

```diff
@@ -4,15 +4,67 @@
 
 import logging
 import os
-
+import datetime
+import boto3
+import time
 from alembic import command
+from alembic.script import ScriptDirectory
+from alembic.migration import MigrationContext
 from alembic.config import Config
+from dataall.base.db.connection import ENVNAME, get_engine
 
 logger = logging.getLogger()
 logger.setLevel(os.environ.get('LOG_LEVEL', 'INFO'))
 
 
 def handler(event, context) -> None:
+    """
+    This function will be called once upon every deployment.
+    It checks if there are any alembic migration scripts to execute.
+    If there are, it will create a snapshot of the database.
+    It executes the alembic migration scripts.
+    """
     alembic_cfg = Config('alembic.ini')
     alembic_cfg.set_main_option('script_location', './migrations')
+
+    # Get head version
+    script = ScriptDirectory.from_config(alembic_cfg)
+    head_rev = script.get_current_head()
+
+    # Get current version from database
+    engine = get_engine(ENVNAME)
+    with engine.engine.connect() as connection:
+        context = MigrationContext.configure(connection)
+        current_rev = context.get_current_revision()
+
+    if head_rev != current_rev:
+        snapshot_id = f'dataall-migration-{head_rev}-{datetime.datetime.now().strftime("%Y-%m-%d-%H-%M-%S")}'
+        cluster_id = engine.dbconfig.host.split('.')[0]
+        logger.info(
+            f'Creating RDS snapshot for cluster {cluster_id}, head revision {head_rev} is ahead of {current_rev}...'
+        )
+        try:
+            rds_client = boto3.client('rds', region_name=os.getenv('AWS_REGION'))
+            cluster_id = 'res-fr-db'
+            cluster_status = ''
+            while cluster_status != 'available':
```

Comment:
nit: I am not saying it should be done like that but take a look at the[ walrus operator](https://nickdrozd.github.io/2022/12/14/walrus-while.html)

Essentially this can be simplified to

```
while (cluster_status := rds_client.describe_db_clusters(DBClusterIdentifier=cluster_id)['DbClusters'][0]['Status'])) is not 'available':
  logger.log(...)
  time.sleep(30)
```

Reply:
Solved in https://github.com/data-dot-all/dataall/pull/1267/commits/e72c76b0c7dd1b95d909e06974995054858defd3